# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/ipc/RemoteVideoDecoder.cpp
# Commit: c633a515314d
# Full Hash: c633a515314d0d1451322d4c6bfda0031cadb459
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2019-06-28 16:12:48
# Regressor Bug: 1561178
# File Overlap Count: 1
# Description:
#   Bug 1561178 - Combine the two video RecvOutput methods and switch based on the descriptor type. r=jya
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D35964
# ==============================================================================

diff -r 0883abbe6a6f -r c633a515314d dom/media/ipc/RemoteVideoDecoder.cpp
--- a/dom/media/ipc/RemoteVideoDecoder.cpp	Fri Jun 28 08:28:53 2019 +0000
+++ b/dom/media/ipc/RemoteVideoDecoder.cpp	Fri Jun 28 06:31:59 2019 +0000
@@ -119,17 +119,33 @@
     const DecodedOutputIPDL& aDecodedData) {
   AssertOnManagerThread();
   MOZ_ASSERT(aDecodedData.type() == DecodedOutputIPDL::TRemoteVideoDataIPDL);
+
   const RemoteVideoDataIPDL& aData = aDecodedData.get_RemoteVideoDataIPDL();
 
-  RefPtr<Image> image = DeserializeImage(
-      aData.sd().get_SurfaceDescriptorBuffer(), aData.frameSize());
+  if (aData.sd().type() == SurfaceDescriptor::TSurfaceDescriptorBuffer) {
+    RefPtr<Image> image = DeserializeImage(
+        aData.sd().get_SurfaceDescriptorBuffer(), aData.frameSize());
+
+    RefPtr<VideoData> video = VideoData::CreateFromImage(
+        aData.display(), aData.base().offset(), aData.base().time(),
+        aData.base().duration(), image, aData.base().keyframe(),
+        aData.base().timecode());
 
-  RefPtr<VideoData> video = VideoData::CreateFromImage(
-      aData.display(), aData.base().offset(), aData.base().time(),
-      aData.base().duration(), image, aData.base().keyframe(),
-      aData.base().timecode());
+    mDecodedData.AppendElement(std::move(video));
+  } else {
+    // The Image here creates a TextureData object that takes ownership
+    // of the SurfaceDescriptor, and is responsible for making sure that
+    // it gets deallocated.
+    RefPtr<Image> image =
+        new GPUVideoImage(GetManager(), aData.sd(), aData.frameSize());
 
-  mDecodedData.AppendElement(std::move(video));
+    RefPtr<VideoData> video = VideoData::CreateFromImage(
+        aData.display(), aData.base().offset(), aData.base().time(),
+        aData.base().duration(), image, aData.base().keyframe(),
+        aData.base().timecode());
+
+    mDecodedData.AppendElement(std::move(video));
+  }
   return IPC_OK();
 }
 
@@ -188,27 +204,6 @@
 GpuRemoteVideoDecoderChild::GpuRemoteVideoDecoderChild()
     : RemoteVideoDecoderChild(true) {}
 
-mozilla::ipc::IPCResult GpuRemoteVideoDecoderChild::RecvOutput(
-    const DecodedOutputIPDL& aDecodedData) {
-  AssertOnManagerThread();
-  MOZ_ASSERT(aDecodedData.type() == DecodedOutputIPDL::TRemoteVideoDataIPDL);
-  const RemoteVideoDataIPDL& aData = aDecodedData.get_RemoteVideoDataIPDL();
-
-  // The Image here creates a TextureData object that takes ownership
-  // of the SurfaceDescriptor, and is responsible for making sure that
-  // it gets deallocated.
-  RefPtr<Image> image =
-      new GPUVideoImage(GetManager(), aData.sd(), aData.frameSize());
-
-  RefPtr<VideoData> video = VideoData::CreateFromImage(
-      aData.display(), aData.base().offset(), aData.base().time(),
-      aData.base().duration(), image, aData.base().keyframe(),
-      aData.base().timecode());
-
-  mDecodedData.AppendElement(std::move(video));
-  return IPC_OK();
-}
-
 void GpuRemoteVideoDecoderChild::RecordShutdownTelemetry(
     bool aAbnormalShutdown) {
 #ifdef XP_WIN